control-plane/connect-inject/endpoints_controller_test.go

```diff
@@ -202,6 +202,164 @@ func TestProcessUpstreams(t *testing.T) {
 		consulNamespacesEnabled bool
 		consulPartitionsEnabled bool
 	}{
+		{
```

Comment:
Do we test the error cases here? Should we add tests to validate the behavior on a few error cases?

Reply:
Added tests for error cases, great callout!